### ISSUE TITLE
Can yolov3 tiny be used for face landmark estimation ?

### ISSUE THREAD

**Issue by uday60** [AlexeyAB/darknet#1141]
I was wondering whether I can train yolov3 for 68 face landmark ? Do I need to make changes in any of files for this ?

Thanks.

**Comment by AlexeyAB** [comment]
Landmark is a point (x,y), but Yolo by default can detect only bounded boxes (x,y,w,h).

You can try to mark landmark as bounded boxes - x,y of landmark and constant w=0.01 h=0.01
And the train it.

----

But it is better to use special approaches for this particular task:

* https://github.com/ShiqiYu/libfacedetection
* https://arxiv.org/pdf/1511.07212v1.pdf
* https://www.ri.cmu.edu/pub_files/2013/5/main.pdf
* http://7xrqgw.com1.z0.glb.clouddn.com/sdmppt.pdf

----

Or you should change source code in many places, that it can get not only 4 coords (x,y,w,h) but also `2*landmarks` coords, here:

*  everywhere there are value `4` in this file: https://github.com/AlexeyAB/darknet/blob/04c0c4fdf84bb7b9e89604e8b9a6079016d5968c/src/yolo_layer.c

* https://github.com/AlexeyAB/darknet/blob/04c0c4fdf84bb7b9e89604e8b9a6079016d5968c/src/yolo_layer.c#L84-L109

* https://github.com/AlexeyAB/darknet/blob/04c0c4fdf84bb7b9e89604e8b9a6079016d5968c/src/data.c#L293-L373

**Comment by uday60** [comment]
I was hoping I would be able to mark face and 68 landmark in the same training model. As you know my live face detection didn't work last time so now I have to resort to faster models. So can I mark the face and landmark in the same training model ?

**Comment by AlexeyAB** [comment]
Yes, you can , but you should do many changes in the source code.

Or just use this software: https://github.com/ShiqiYu/libfacedetection

**Comment by uday60** [comment]
@AlexeyAB 
This doesn't work on Linux.

**Comment by uday60** [comment]
If I enter the x,y coordinates with width 1 and height 0 for 68 landmarks. Would it work together with face detection ?

**Comment by uday60** [comment]
I have another question related to minimum size. How can i set the minimum size of the object ?
I am only looking to detect objects greater than 40x40.
Also while running yolov3 tiny i noticed that it only took about 400 mb of GPU memory and was able to handle HD video with 20 FPS. Why is it not taking more memory when available to speed up ? Can GPU memory be changed or extended via command line ?

Also what yolov3 didn't take more GPU memory and stopped at 1.3 GB. What is going on ?

**Comment by AlexeyAB** [comment]
> I have another question related to minimum size. How can i set the minimum size of the object ?

What do you mean?

> I am only looking to detect objects greater than 40x40.

You don't want to detect object smaller than 40x40, isn't it?  Why?

> Also while running yolov3 tiny i noticed that it only took about 400 mb of GPU memory and was able to handle HD video with 20 FPS. Why is it not taking more memory when available to speed up ? Can GPU memory be changed or extended via command line ?
> 
> Also what yolov3 didn't take more GPU memory and stopped at 1.3 GB. What is going on ?

Just Yolo doesn't require more GPU memory.
GPU-memory size isn't related to the performance. Performance depends on the number of GPU-Cores and on GPU-Clock (MHz) rather than GPU-memory size.

**Comment by ouyanghuiyu** [comment]
I modified some files for face detection (with landmark) : https://github.com/ouyanghuiyu/darknet_face_with_landmark
